packages/runtime/src/core.ts

```diff
@@ -517,13 +518,30 @@ export class FederationHost {
         moduleInstance: module,
         origin: this,
       });
+      //@ts-ignore
+      console.log(
+        'moduleOrFactory',
+        moduleOrFactory?.toString(),
+      //@ts-ignore
+        moduleOrFactory(),
+      );
+
       return moduleOrFactory;
     } catch (error) {
-      this.hooks.lifecycle.errorLoadRemote.emit({
+      const { from = 'runtime' } = options || { from: 'runtime' };
+
+      const failover = await this.hooks.lifecycle.errorLoadRemote.emit({
         id,
```

Comment:
Should we add origin to errorLoadRemote hook params ? Because we can use this to re-call origin.loadRemote to implement retry plugin @zhoushaw

Reply:
yes i think we should make hook be able to return args for loadRemoteHook, maybe with some extra data, but then we can apply the same lifecycle to failovers. 

For example, with that - i could use beforeLoadRemote({wasOffline})
or something, then i could either insert new remoteEntry, or mock one in module cache, or just do what i do currently, and return mock Module